src/components/Api.js

```diff
@@ -4,25 +4,35 @@ export default class Api {
     this._header = url.headers;
   }
 
-  _checkRequest(res) {
+  async _checkRequest(res) {
     if (res.ok) {
-      return res.json();
+      return await res.json();
     } else {
       return Promise.reject(`Error: ${res.status}`);
     }
   }
 
-  _request(url, options) {
-    return fetch(url, options).then(this._checkRequest);
+  async _request(url, options) {
+    try {
+      const res = await fetch(url, options);
+      return this._checkRequest(res);
+    } catch (error) {
```

Comment:
There is no need to add `catch` block here, since you should handle errors in index.js